### PR TITLE
fix for https://github.com/cpitclaudel/company-coq/issues/241

### LINKED ISSUE #241: `M-.` and notations
Consider the following code:

```
Definition foo := unit.
Notation bar := foo.
Lemma FooBar: foo = bar.
```

In edit buffer positioning the cursor on `bar` in the goal and pressing `M-.` jumps to the definition of `bar` notation.
Doing the same in the `*goals*` buffer triggers an error: "No location found for 'bar'".

Coq 8.11.2 (via opam)
GNU Emacs 26.3 (build 2, x86_64-pc-linux-gnu, GTK+ Version 3.22.30) of 2019-09-16
Compan-coq Version: 20200625.1650 (via MELPA)

### PR DESCRIPTION
**Pull request by vzaliva**
Fixes #241 per @cpitclaudel suggestion

### CODE REVIEW

File: company-coq.el [hunk 1]
```diff
@@ -2174,9 +2174,10 @@ to a non-existent file (for an example of such a case, try
 \[Locate Library Peano] in 8.4pl3)."
   (if (and (equal lib-path "")
            (or (equal mod-name "Top")
-               (and (buffer-file-name proof-script-buffer)
-                    (equal (concat mod-name ".v")
-                           (file-name-nondirectory (buffer-file-name proof-script-buffer))))))
+               (and proof-script-buffer
+                    (and (buffer-file-name proof-script-buffer)
```
Comment (cpitclaudel) [inline company-coq.el:2178]: Perfect. Since `and` can take any number of arguments there's no need to nest though, you can have just one `and` call.

Comment (vzaliva) [reply company-coq.el:2178]: is it guaranteed that the conditions will be checked from left to right and it will stop if the first one is false?

Comment (cpitclaudel) [reply company-coq.el:2178]: Thankfully yes

**Comment by cpitclaudel** [comment]
Thanks, squashed and merged with a small edit.  Not sure how to tell github that it was in fact merged, but I checked that the authorship was preserved.

**Pull request closed without merging**